astropy/io/fits/convenience.py

```diff
@@ -709,6 +712,110 @@ def info(filename, output=None, **kwargs):
+    """
+    Compare two parts of a fits file, including entire FITS files, and
+    FITS HDU objects.
+
+    Parameters
+    ----------
+    inputa : file path, file object, or file like object
+        The filename of a FITS file on disk, or fits file object
+        to compare to inputb.
+
+    inputb : file path, file object, or file like object
+        The filename of a FITS file on disk, or a fits file object
+        to compare to inputa.
+
+    ext
+        The rest of the arguments are for extension specification if your
+        inputA and inputB are string filenames (will not work if inputA and
+        inputB are HDU objects. They are flexible and are best illustrated
+        by examples.
+
+        By extension number::
+
+            quickdiff('inA.fits', 'inB.fits', 0)      # the primary HDU
+            quickdiff('inA.fits', 'inB.fits', 2)      # the second extension
+            quickdiff('inA.fits', 'inB.fits', ext=2)  # the second extension
+
+        By name, i.e., ``EXTNAME`` value (if unique)::
+
+            quickdiff('inA.fits', 'inB.fits', 'sci')
+            quickdiff('inA.fits', 'inB.fits', extname='sci')  # equivalent
+
+        Note ``EXTNAME`` values are not case sensitive
+
+        By combination of ``EXTNAME`` and ``EXTVER`` as separate
+        By combination of ``EXTNAME`` and ``EXTVER`` as separate
+        arguments or as a tuple::
+
+            quickdiff('inA.fits', 'inB.fits', 'sci', 2)  # EXTNAME='SCI' & EXTVER=2
+            quickdiff('inA.fits', 'inB.fits', extname='sci', extver=2)  # equivalent
+            quickdiff('inA.fits', 'inB.fits', ('sci', 2))  # equivalent
+
+        Ambiguous or conflicting specifications will raise an exception::
+
+            quickdiff('inA.fits', 'inB.fits', ext=('sci',1), extname='err', extver=2)
+
+
+    kwargs
+        Any additional keyword arguments to be passed to
+        `astropy.io.fits.FITSDiff`.
+
+    Notes
+    -----
+    The primary use for the `quickdiff` function is to allow quick print out
+    of a FITS difference report and will write to ``sys.stdout``.
+    To save the diff report to a file please use `astropy.io.fits.FITSDiff`
+    directly.
+    """
+
+    if isinstance(inputa, string_types) and isinstance(inputb, string_types):
+
+        # Use ext keywords if filename strings provided, unless none exist, then run FITSDIFF directly
+        if len(args) == 0 and 'ext' not in kwargs and 'extname' not in kwargs and 'extver' not in kwargs:
+            print(FITSDiff(inputa, inputb, **kwargs).report())
+
+        else:
+            # Use handy _getext to interpret any ext keywords
+            modea, closeda = _get_file_mode(inputa)
+            modeb, closedb = _get_file_mode(inputb)
+            hdulista, extidxa = _getext(inputa, modea, *args, **kwargs)
+            hdulistb, extidxb = _getext(inputa, modeb, *args, **kwargs)
+
+            # Pop those extension keywords if they exist
+            kwargs.pop('ext', None)
```

Comment:
It doesn't seem to be successfully doing this... and i'm not sure why.  Although in my use case this works in my favor, since I pass the same **kwarg set to the second file as well.  I don't actually see _getext returning the **kwarg list, is it supposed to be "passed by reference"?  Trying to google this and not quite sure how the change from a list of keywords to the collected dictionary version that gets popped works in that respect.